packages/testsuite/cypress/e2e/undertow/test-configuration-subsystem-undertow-global-settings.cy.ts

```diff
@@ -0,0 +1,114 @@
+
+        cy.executeInWildflyContainer(`/subsystem=undertow/server=new-server/host=${testValues.serverHost}:add`);
+      });
+  });
+
+  after(() => {
+    cy.task("stop:containers");
+  });
+
+  beforeEach(() => {
+    cy.navigateTo(managementEndpoint, "undertow");
+    // the form takes a brief moment to initialize
+    cy.wait(200);
+  });
+
+  it("Should update the default security domain configuration ", () => {
+    cy.editForm(globalSettingsForm.formId);
+    cy.text(globalSettingsForm.formId, globalSettingsForm.defaultSecurityDomain, "some test value");
+    cy.saveForm(globalSettingsForm.formId);
+    cy.verifySuccess();
+    cy.verifyAttribute(managementEndpoint, address, "default-security-domain", "some test value");
+  });
+
+  it("Should update the default server configuration", () => {
+    cy.editForm(globalSettingsForm.formId);
+    cy.text(globalSettingsForm.formId, globalSettingsForm.defaultServer, testValues.serverName);
+    cy.saveForm(globalSettingsForm.formId);
+    cy.verifySuccess();
+    cy.verifyAttribute(managementEndpoint, address, "default-server", testValues.serverName);
+  });
+
+  it("Should update the default servlet container configuration", () => {
+    cy.editForm(globalSettingsForm.formId);
+    cy.text(globalSettingsForm.formId, globalSettingsForm.defaultServletContainer, testValues.servletContainer);
+    cy.saveForm(globalSettingsForm.formId);
+    cy.verifySuccess();
+    cy.verifyAttribute(managementEndpoint, address, "default-servlet-container", testValues.servletContainer);
+  });
+
+  it("Should update the default virtual host configuration", () => {
+    cy.editForm(globalSettingsForm.formId);
+    cy.text(globalSettingsForm.formId, globalSettingsForm.defaultVirtualHost, testValues.serverHost);
+    cy.saveForm(globalSettingsForm.formId);
+    cy.verifySuccess();
+    cy.verifyAttribute(managementEndpoint, address, "default-virtual-host", testValues.serverHost);
+  });
+
+  it("Should update the instance ID configuration", () => {
+    cy.editForm(globalSettingsForm.formId);
+    cy.text(globalSettingsForm.formId, globalSettingsForm.instanceId, testValues.instanceId);
+    cy.saveForm(globalSettingsForm.formId);
+    cy.verifySuccess();
+    cy.verifyAttribute(managementEndpoint, address, "instance-id", testValues.instanceId);
+  });
+
+  it("Should update the statistics-enabled configuration by changing the expression", () => {
+    console.log("forgodssake");
+    cy.editForm(globalSettingsForm.formId);
+    const expressionFormInput = "input#undertow-global-settings-form-statistics-enabled-editing.form-control";
+    cy.get(expressionFormInput).click().wait(200).clear();
+    cy.get(expressionFormInput)
+      .type(testValues.statisticsEnabledExpression, {
+        parseSpecialCharSequences: false,
+      })
+      .type("{enter}");
+
+    cy.get(expressionFormInput).trigger("change");
```

Comment:
Could this be somehow made into a function for example in `form-editing.ts`? I believe that it could be helpful in other cases

Reply:
Does it work also from false to true?

Reply:
I'm not sure what you mean by false/true. The possible states of the statistics-enabled are`"expression"`, `true`, `false`.
The default state is `"expession"`. This test tries to change the default expression to another expression. This is possible only by using the `"parseSpecialCharSequences: false"` option for `type()`, so as not to iterpret `{text}` as a special command.

Reply:
I've refactored the input of expression into `cy.textExpression()` function and the verification into `cy.verifyAttributeAsExpression()`. I've also added an option parameter to `cy.text()`, it now takes an object as an optional 4th parameter, with options to specify your own selector and to opt out of parsing of special characters.
This does not change the behaviour of `cy.text()`, but adds a feature.
`cy.textExpression()` then builds onto `cy.text()`.